solver/src/settlement/settlement_encoder.rs

```diff
@@ -208,6 +208,20 @@ impl SettlementEncoder {
         })
     }
 
+    pub fn total_fees(&self) -> Option<U256> {
+        self.trades.iter().fold(Some(0.into()), |acc, trade| {
+            let fees = trade.order.order_creation.fee_amount;
+            Some(acc? + fees)
+        })
+    }
+
+    pub fn total_unsubsidized_fees(&self, fee_discount_factor: f64) -> Option<U256> {
```

Comment:
If you wanted to save some boilerplate you could get rid of this method and just divide total_fees by the fee_discount_factor in the layer above (but not a huge difference).

Reply:
Maybe a small unit test would be nice?